internal/kallsyms/kallsyms_test.go

```diff
@@ -0,0 +1,34 @@
+package kallsyms
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/go-quicktest/qt"
+)
+
+func TestKernelModule(t *testing.T) {
+	kallsyms := []byte(`0000000000000000 t hid_generic_probe	[hid_generic]
```

Comment:
Repeating my earlier comment: https://github.com/cilium/ebpf/pull/1300#discussion_r1453494363

According to https://stackoverflow.com/questions/39120818/what-is-the-difference-between-t-and-t-in-proc-kallsyms and `man nm` my /proc/kallsyms also includes symbols which are not functions. Why not filter those out?